src/machine/machined-varlink.c

```diff
@@ -508,6 +509,118 @@ static int vl_method_terminate(sd_varlink *link, sd_json_variant *parameters, sd
         return lookup_machine_and_call_method(link, parameters, flags, userdata, vl_method_terminate_internal);
 }
 
+static int list_image_one_and_maybe_read_metadata(sd_varlink *link, Image *image, bool more, bool read_metadata) {
```

Comment:
I'd like to move this and below to image-varlink.c

Reply:
I don't have a strong opinion, but the same DBus logic is in machined-dbus.c. So, I copied it here to be consistent. I understand that DBus has per-object methods, which make a situation a bit different. Yet, I'd keep it here. Or we would also need to move `list_machine_one_and_maybe_read_metadata()` to machine-varlink.c for consistency. If so, I better do it in a separate PR.

Reply:
OK, feel free to keep them. We can move them later.

Reply:
I've realized that `list_machine_one_and_maybe_read_metadata()` is from upcoming PR. Just FYI.